provider/cluster_upgrade.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright (c) 2023 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package provider
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	"github.com/openshift/rosa/pkg/ocm"
+)
+
+// clusterUpgrade bundles the description of the upgrade with its current state
+type clusterUpgrade struct {
+	policy      *cmv1.UpgradePolicy
+	policyState *cmv1.UpgradePolicyState
+}
+
+func (cu *clusterUpgrade) State() cmv1.UpgradePolicyStateValue {
+	return cu.policyState.Value()
+}
+
+func (cu *clusterUpgrade) Version() string {
+	return cu.policy.Version()
+}
+
+func (cu *clusterUpgrade) NextRun() time.Time {
+	return cu.policy.NextRun()
+}
+
+func (cu *clusterUpgrade) Delete(ctx context.Context, client *cmv1.ClustersClient) error {
+	_, err := client.Cluster(cu.policy.ClusterID()).UpgradePolicies().UpgradePolicy(cu.policy.ID()).Delete().SendContext(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to delete upgrade policy: %v", err)
+	}
+	return nil
+}
+
+// Get the available upgrade versions that are reachable from a given starting
+// version
+func getAvailableUpgrades(ctx context.Context, client *cmv1.VersionsClient, fromVersionId string) ([]*cmv1.Version, error) {
```

Comment:
```suggestion
func getAvailableUpgradeVersions(ctx context.Context, client *cmv1.VersionsClient, fromVersionId string) ([]*cmv1.Version, error) {
```
To make it more clear that the function return versions and not ClusterUpgrade